signature/internal/signature/mechanism.go

```diff
@@ -0,0 +1,34 @@
+package signature
```

Comment:
This can be just `signature/internal`, and avoid the `internalSignature` naming.

---

… But, actually, that still makes the interface only reachable by the `c/image/signature` namespace (e.g. not from `c/image/copy`). So I don’t think we need an internal sub package at all — just name the private interface using a private symbol, e.g. `c/image/signature.signingMechanismWithPassphrase`.

IIRC that also makes the documentation better: a type alias in the public package to the internal package makes the interface _usable_ but `go doc` etc. can’t follow it and don’t show the reachable public methods. So keeping `c/image/signature.SigningMechanism` in place, and using a private interface in the same subpackage, would work a bit better.

OTOH, I could well accept the argument that we want to deprecate any external uses of `SigningMechanism` anyway, and hading the methods from documentation is actually a feature :) In that case an explicit comment that the interface is deprecated (like `types.BlobInfoCache`) might be useful.

Reply:
Went with `c/image/signature.signingMechanismWithPassphrase`